packages/opencensus-instrumentation-redis/src/redis.ts

```diff
@@ -0,0 +1,150 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import {BasePlugin, Span, SpanKind, CanonicalCode} from '@opencensus/core';
+import * as shimmer from 'shimmer';
+import * as semver from 'semver';
+
+
+/** Redis instrumentation plugin for Opencensus */
+export class RedisPlugin extends BasePlugin {
+  /** Constructs a new Redis instance. */
+  constructor(moduleName: string) {
+    super(moduleName);
+  }
+
+  /**
+   * Patches Redis operations.
+   */
+  protected applyPatch() {
+    this.logger.debug('Patched redis');
+
+    if (semver.lt(this.version, '2.6.0')) {
+      this.logger.info('disabling redis plugin because version isnt supported');
+      return this.moduleExports;
+    }
+
+    if (this.moduleExports.RedisClient) {
+      this.logger.debug('patching redis.RedisClient.prototype.create_stream');
+      shimmer.wrap(
+          this.moduleExports.RedisClient.prototype, 'create_stream',
+          this.getPatchCreateStream());
+      this.logger.debug('patching redis.RedisClient.prototype.internal_send');
+      shimmer.wrap(
+          this.moduleExports.RedisClient.prototype, 'internal_send_command',
+          this.getPatchSendCommand());
+      this.logger.debug('patching redis.RedisClient.prototype.createClient');
+      shimmer.wrap(
+          this.moduleExports, 'createClient', this.getPatchCreateClient());
+    }
+    return this.moduleExports;
+  }
+
+  /** Unpatches all Redis patched functions. */
+  applyUnpatch(): void {
+    if (semver.lt(this.version, '2.6.0')) return;
+
+    shimmer.unwrap(
+        this.moduleExports.RedisClient.prototype, 'internal_send_command');
+    shimmer.unwrap(this.moduleExports, 'createClient');
+    shimmer.unwrap(this.moduleExports.RedisClient.prototype, 'create_stream');
+  }
+
+  /** Patch internal event emmitter to propagate the context */
+  private getPatchCreateStream() {
+    const plugin = this;
+    return function createStreamWrap(original: Function) {
+      // tslint:disable-next-line:no-any
+      return function create_stream_trace(this: any) {
+        if (!this.stream) {
+          Object.defineProperty(this, 'stream', {
+            get() {
+              return this._patched_redis_stream;
+            },
+            set(val) {
+              plugin.tracer.wrapEmitter(val);
+              this._patched_redis_stream = val;
+            }
+          });
+        }
+        return original.apply(this, arguments);
+      };
+    };
+  }
+
+  /** Patch client event emmitter to propagate the context */
+  private getPatchCreateClient() {
+    const plugin = this;
+    return function createClientWrap(original: Function) {
+      // tslint:disable-next-line:no-any
+      return function createClientTrace(this: any) {
```

Comment:
Similar comment here on the `any`. Can you use the actual type or create a type if needed for it?